src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/DefaultOpenDistroSecurityKeyStore.java

```diff
@@ -223,313 +229,374 @@ private void initSSLConfig() {
+                final X509Certificate[] httpKeystoreCert = new X509Certificate[]{ PemKeyReader.loadCertificateFromFile(pemCertFilePath) };
+                validateNewCerts(httpCerts, httpKeystoreCert);
+                setHttpSSLCerts(httpKeystoreCert);
+                httpSslContext = buildSSLServerContext(new File(pemKey), new File(pemCertFilePath),
+                    trustedCas == null ? null : new File(trustedCas),
+                    settings.get(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_PASSWORD),
+                    getEnabledSSLCiphers(this.sslHTTPProvider, true), sslHTTPProvider, httpClientAuthMode);
+            } catch (final Exception e) {
+                logExplanation(e);
+                throw new ElasticsearchSecurityException(
+                    "Error while initializing http SSL layer from PEM: " + e.toString(), e);
+            }
 
-                    checkPath(trustedCas, SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMTRUSTEDCAS_FILEPATH);
+        } else {
+            throw new ElasticsearchException(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_KEYSTORE_FILEPATH + " or "
+                + SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_FILEPATH
+                + " must be set if http ssl is reqested.");
+        }
+    }
 
-                }
 
-                final String pemCertFilePath = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMCERT_FILEPATH, true);
-                final String pemKey = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_FILEPATH, true);
+    /**
+     * For new X509 cert to be valid Issuer, Subject DN must be the same and
+     * new certificates should expire after current ones.
+     * @param currentX509Certs  Array of current x509 certificates
+     * @param newX509Certs      Array of x509 certificates which will replace our current cert
+     * @throws Exception if certificate is invalid
+     */
+    private void validateNewCerts(final X509Certificate[] currentX509Certs, final X509Certificate[] newX509Certs) throws Exception {
 
-                try {
-                    httpSslContext = buildSSLServerContext(new File(pemKey), new File(pemCertFilePath),
-                            trustedCas == null ? null : new File(trustedCas),
-                            settings.get(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_PASSWORD),
-                            getEnabledSSLCiphers(this.sslHTTPProvider, true), sslHTTPProvider, httpClientAuthMode);
-                } catch (final Exception e) {
-                    logExplanation(e);
-                    throw new ElasticsearchSecurityException(
-                            "Error while initializing http SSL layer from PEM: " + e.toString(), e);
-                }
+        // First time we init certs ignore validity check
+        if (currentX509Certs == null) {
```

Comment:
What if currentX509Certs is an empty list ?

Reply:
I've added back the empty list check. Current x509 certs array will always need to have an array of values.